numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -329,6 +336,93 @@ array_implement_array_function(
 }
 
 
+/*
+ * Implements the __array_function__ protocol for a Python function, as described in
+ * in NEP-18. See numpy.core.overrides for a full docstring.
+ */
+NPY_NO_EXPORT PyObject *
+array_implement_array_function(
+    PyObject *NPY_UNUSED(dummy), PyObject *positional_args)
+{
+    PyObject *implementation, *public_api, *relevant_args, *args, *kwargs;
+
+    if (!intern_strings())
+        return NULL;
+
+    if (!PyArg_UnpackTuple(
+            positional_args, "implement_array_function", 5, 5,
+            &implementation, &public_api, &relevant_args, &args, &kwargs))
+        return NULL;
+
+    /* Remove `like=` kwarg, which is NumPy-exclusive and thus not present
```

Comment:
I'm not sure what you mean by that, what are "most cases"? Note that removing the `like=` keyword is part of [NEP-35](https://numpy.org/neps/nep-0035-array-creation-dispatch-with-array-function.html#implementation), if we intend not to do that, we should probably discuss it and change the NEP as well.

Reply:
Well, this also removes `like=` from `np.concatenate((arr1, arr2), out=out)` which clearly doesn't support `like` I think? I do agree that it is unlikely it is speed relevant though.

Reply:
I see your point now. On your previous comment you mentioned "it cannot even be present", are you referring that `like=` cannot be present or that this code block cannot be present?

Reply:
Sorry, yeah, I just meant that many functions will never have a `like` kwarg, since they do not support it. Maybe would be good to just have a comment that we currently remove it even for examples like concatenate.